src/Data/FastVect/Common/Matrix.purs

```diff
@@ -0,0 +1,160 @@
+module Data.FastVect.Common.Matrix where
+
+import Prelude
+
+import Data.FastVect.Common (NegOne, One, Zero)
+import Data.Reflectable (class Reflectable)
+import Prim.Int (class Add, class Compare)
+import Prim.Ordering (GT, LT)
+import Type.Proxy (Proxy)
+
+type Index matrix h h_minus_one w w_minus_one i j m n elem =
```

Comment:
I think with respect to your comment here https://github.com/sigma-andex/purescript-fast-vect/issues/17
this could then also be simplified